R/testdata.R

```diff
@@ -55,91 +55,123 @@ dataset_test_worker <-
 
     # We're using 2nd edition of test that, which has "context" field
     # https://cran.r-project.org/web/packages/testthat/vignettes/third-edition.html
+    # Is there a reason to be using the 2nd edition, while context has been superseded in later versions?
     local_edition(2)
 
-    ## New expect_that helper functions; test that a number is in a range,
-    ## or that a range contains a number.
-
-    expect_isin <-
-      function(object,
-               expected,
-               ...,
-               info = NULL,
-               label = NULL,
-               expected.label = NULL,
-               na.rm = TRUE) {
+    expect_is_in <- function(object, expected, ..., info, label, na.rm = TRUE) {
+
         if (na.rm)
           object <- object[!is.na(object)]
         i <- object %in% expected
 
         comp <- compare(all(i), TRUE, ...)
-        expect(comp$equal,
-               sprintf(
-                 "%s - should not contain: %s",
-                 info,
-                 paste(object[!i], collapse = ", ")
-               ))
+        expect(
+          comp$equal,
+          sprintf(
+            "%s - %s should not contain: %s",
+            info, label,
+            paste(object[!i], collapse = ", ")
+          ))
 
         invisible(object)
       }
 
-    # Expectation: one set contains the other
-    expect_contains <- function(object, expected, ..., info = NULL) {
+    expect_contains <- function(object, expected, ..., info) {
+
       i <- expected %in% object
 
       comp <- compare(all(i), TRUE, ...)
-      expect(comp$equal,
-             sprintf(
-               "%s - does not contain: %s",
-               info,
-               paste(expected[!i], collapse = ", ")
-             ))
+      expect(
+        comp$equal,
+        sprintf("%s - does not contain: %s", info, paste(expected[!i], collapse = ", "))
+      )
 
       invisible(object)
     }
 
-    # Expectation: one set contains the other
-    expect_allowed <- function(object, allowed, ..., info = NULL) {
+    expect_allowed <- function(object, allowed, ..., info, label) {
+
       i <- object %in% allowed
 
       comp <- compare(all(i), TRUE, ...)
-      expect(comp$equal,
-             sprintf(
-               "%s - includes invalid terms: %s",
-               info,
-               paste(object[!i], collapse = ", ")
-             ))
+      expect(
+        comp$equal,
+        sprintf(
+          "%s - %s include(s) invalid terms: %s",
+          info, label,
+          paste(object[!i], collapse = ", ")
+        ))
 
       invisible(object)
     }
 
-    expect_not_NA <- function(object, info = NULL, label = NULL) {
+    expect_equal <- function(object, expected, info) {
+      i <- object == expected
+      comp <- compare(all(i), TRUE)
+      expect(comp$equal, info)
+    }
+
+    expect_true <- function(object, info) {
```

Comment:
Aren't these already included in testthat?

Reply:
Yes, but writing my own versions allow full customisation of failure messages, otherwise there's a lot of fluff in the defaults.